contracts/governance/FeeSharingCollector/FeeSharingCollector.sol

```diff
@@ -393,133 +422,261 @@ contract FeeSharingCollector is
-            wrbtcToken.withdraw(wrbtcAmount);
-        }
-
-        // pull out the iWRBTC to rbtc to this feeSharingCollector contract
-        if (iWrbtcAmount > 0) {
-            processedCheckpoints[user][loanPoolTokenWRBTC] = endIWRBTC;
-            iWRBTCloanAmountPaid = ILoanTokenWRBTC(loanPoolTokenWRBTC).burnToBTC(
-                address(this),
-                iWrbtcAmount,
-                false
-            );
+        (totalAmount, endTokenCheckpoint) = _getRBTCBalance(_token, user, _maxCheckpoints);
+
+        if (totalAmount > 0) {
+            processedCheckpoints[user][_token] = endTokenCheckpoint;
+            if (_token == address(wrbtcToken)) {
+                // unwrap the wrbtc
+                wrbtcToken.withdraw(totalAmount);
+            } else if (_token == loanPoolTokenWRBTC) {
+                // pull out the iWRBTC to rbtc to this feeSharingCollector contract
+                /** @dev will use the burned result from IWRBTC to RBTC as return total amount */
+                totalAmount = ILoanTokenWRBTC(loanPoolTokenWRBTC).burnToBTC(
+                    address(this),
+                    totalAmount,
+                    false
+                );
+            }
         }
 
-        uint256 totalAmount = rbtcAmount.add(wrbtcAmount).add(iWRBTCloanAmountPaid);
-        require(totalAmount > 0, "FeeSharingCollector::withdrawFees: no rbtc for a withdrawal");
-
-        // withdraw everything
-        (bool success, ) = _receiver.call.value(totalAmount)("");
-        require(success, "FeeSharingCollector::withdrawRBTC: Withdrawal failed");
-
         emit RBTCWithdrawn(user, _receiver, totalAmount);
     }
 
     /**
      * @dev withdraw all of the RBTC balance based on particular checkpoints
      *
-     * RBTC balance consists of:
-     * - rbtc balance
-     * - wrbtc balance which will be unwrapped to rbtc
-     * - iwrbtc balance which will be unwrapped to rbtc
+     * This function will withdraw RBTC balance which is passed as _token param, so it could be either of these:
+     * - rbtc balance or
+     * - wrbtc balance which will be unwrapped to rbtc or
+     * - iwrbtc balance which will be unwrapped to rbtc or
+     *
      *
+     * @param _tokens array of either RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT or wrbtc address or iwrbtc address
      * @param _maxCheckpoints  Maximum number of checkpoints to be processed to workaround block gas limit
      * @param _receiver An optional tokens receiver (msg.sender used if 0)
      */
-    function withdrawRBTC(uint32 _maxCheckpoints, address _receiver) external nonReentrant {
-        _withdrawRBTC(_maxCheckpoints, _receiver);
+    function withdrawRbtcTokens(
+        address[] calldata _tokens,
+        uint32 _maxCheckpoints,
+        address _receiver
+    ) external nonReentrant {
+        validRBTCBasedTokens(_tokens);
+
+        if (_receiver == ZERO_ADDRESS) {
+            _receiver = msg.sender;
+        }
+
+        uint256 tempRbtcAmountToSend;
```

Comment:
```suggestion
        uint256 rbtcAmountToSend;
```

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/507/commits/efb8c7eef560dc995d949234265110c21647bded)